src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java

```diff
@@ -87,7 +89,20 @@
         @SuppressWarnings("unchecked") // untypable
         @Override protected Void run() throws Exception {
             Job<?,?> job = build.getParent();
-            boolean isMultibranch = isMultibranch(job);
+
+            Run<?,?> previousRun = build.getPreviousCompletedBuild();
```

Comment:
What happens if we are doing concurrent builds? Can things get out of whack?

And what about when all historical builds are deleted as part of some routine cleanup?

May make more sense to store the action on the `Job` rather than on the `Run`, not sure.